### PR TITLE
Abstract Navbar + Livesite pages

### PR DESCRIPTION
**Pull request by ianmah**
- I made the navbar abstract so that livesite can use it too
- Added livesite subpages

image

### CODE REVIEW

File: constants.js
Comment (jvssicawu) [inline constants.js:36]: typo here

File: constants.js
Comment (jvssicawu) [inline constants.js:38]: hmm we might need a FAQ and sponsors (for the category prizes as well as the logos) tab too unless yall are just gonna reroute it from the respective hackathon's FAQ/sponsors and manually add the category prizes

File: pages/Livesite/announcements.js
Comment (jvssicawu) [inline pages/Livesite/announcements.js:16]: p a n i k

**Review by jvssicawu — COMMENTED** [review]
left a suggestion but otherwise lgtm 🤟

File: constants.js
Comment (ianmah) [reply constants.js:38]: Yeah I think we will be doing something like grabbing it from the active hackathon. Maybe we should add a settings page in the livesite area to modify which is the current 'active' hackathon?

File: components/sidebar.js
Comment (jackyzha0) [inline components/sidebar.js:109]: why did this change to intro?

Comment (jackyzha0) [reply components/sidebar.js:109]: shouldn't `getTrailingPath()` return `/intro` when there is no trailing path anyways?

File: components/sidebar.js
Comment (jackyzha0) [inline components/sidebar.js:129]: curious about this capitalization change

File: pages/[id]/intro.js
Comment (jackyzha0) [inline pages/[id]/intro.js:163]: is there a way to add this `Page` to all the pages using a wrapper or container or something? If we end up having to change the props for this, that's a lot of places we're gonna need to change

File: components/sidebar.js
Comment (ianmah) [reply components/sidebar.js:109]: problem is that `Livesite/announcements` returns `announcements` as trailing path, and there is no `[id]/announcements`

File: components/sidebar.js
Comment (ianmah) [reply components/sidebar.js:129]: Looks prettier here LMAO
https://github.com/nwplus/CMS-V2/blob/4e2cc1e103d65a49446520e74f4833c19b5aa6d3/components/page.js#L50

File: pages/[id]/intro.js
Comment (ianmah) [reply pages/[id]/intro.js:163]: 🤔 good idea

File: components/sidebar.js
Comment (Ciuffi) [reply components/sidebar.js:109]: wouldn't this remove the feature of navigating to a different hackathon but staying on the same section?

Comment (Ciuffi) [reply components/sidebar.js:109]: why not just check if the currentpath is Livesite and return a intro or something in the getTrailingPath function?

Comment (ianmah) [reply components/sidebar.js:109]: oh word didn't consider that. Also why is that the functionality? Seems weird to me

Comment (Ciuffi) [reply components/sidebar.js]: Its mostly a feature for the sponsorship team. If they're switching hackathons, it usually means they need to add sponsors to multiple hackathons, so it removes a few clicks for them. For our logs teams, they probably will never switch hackathons, so its shouldn't effect them.

**Review by Ciuffi — APPROVED** [review]
Looks good to me, I'll wait until this is merged to merge my faq changes since it looks like you've made a lot of changes to page/sidebar

**Review by andykwan1124 — APPROVED** [review]
nice, lgtm

**Pull request merged**
